notify/grafana_alertmanager.go

```diff
@@ -184,39 +202,75 @@ func (c *GrafanaAlertmanagerConfig) Validate() error {
 		return errors.New("notification log maintenance options must be present")
 	}
 
+	if c.EmailSender == nil {
+		return errors.New("email sender must be present")
+	}
+
+	if c.ImageProvider == nil {
+		return errors.New("image provider must be present")
+	}
+
+	if c.Decrtypter == nil {
+		return errors.New("decrtypter must be present")
+	}
+
+	if c.LoggerFactory == nil {
+		return errors.New("logger factory must be present")
+	}
+
+	if c.TenantKey == "" {
+		return errors.New("tenant key must be present")
+	}
+
+	if c.Peer == nil {
+		return errors.New("peer must be present")
+	}
+
+	if c.Logger == nil {
+		return errors.New("logger must be present")
+	}
+
+	if c.Metrics == nil {
+		return errors.New("metrics must be present")
+	}
+
 	return nil
 }
 
 // NewGrafanaAlertmanager creates a new Grafana-specific Alertmanager.
-func NewGrafanaAlertmanager(tenantKey string, tenantID int64, config *GrafanaAlertmanagerConfig, peer ClusterPeer, logger log.Logger, m *GrafanaAlertmanagerMetrics) (*GrafanaAlertmanager, error) {
-	// TODO: Remove the context.
+func NewGrafanaAlertmanager(opts GrafanaAlertmanagerOpts) (*GrafanaAlertmanager, error) {
```

Comment:
One nit about the naming is that opts usually refers to the opts functional pattern, which might be a good fit here actually. e.g.;
```suggestion
func NewGrafanaAlertmanager(opts ...func(*GrafanaAlertmanagerOpts)) (*GrafanaAlertmanager, error) {
    amOpts := GrafanaAlertmanagerOpts{
        // set some default values
    }
    for _, o := range opts {
        o(amOpts)
    }
```

`GrafanaAlertmanagerOpts` could be made private also, so we abstract the main configuration from the interface and possibly simplify things a bit more.. not sure it's worth the effort at all, just an idea 👍

Reply:
I called it opts because there was too many "settings" for the same struct. I hesitated to use that pattern because it would bring some unnecessary flexibility and would assume implementation of the factory functions for each field of the options struct to complete it. 
On the second though, it could make it clearer the setting of options.

Reply:
I decided to not go with this as this pattern assumes that fields have sensible defaults. In our case they're not. So, the usage will become pretty verbose.